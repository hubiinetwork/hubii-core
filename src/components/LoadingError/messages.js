/*
 * LoadingError Messages
 *
 * This contains all the text for the LoadingError component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  unknownError: {
    id: 'app.containers.LoadingError.unknownError',
    defaultMessage: 'Error occured: {error}',
  },
  doesntExist: {
    id: 'app.containers.LoadingError.doesntExist',
    defaultMessage: '{type} {id} doesn\'t exist',
  },
});
