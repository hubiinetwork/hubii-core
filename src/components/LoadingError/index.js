/**
*
* LoadingError
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function LoadingError(props) {
  if (props.error.message === 'Not Found') {
    return (
      <h1>
        <FormattedMessage
          {...messages.doesntExist}
          values={{ type: props.pageType, id: props.id }}
        />
      </h1>
    );
  }

  return (
    <h1>
      <FormattedMessage
        {...messages.unknownError}
        values={{ error: props.error.message }}
      />
    </h1>
  );
}

LoadingError.propTypes = {
  error: PropTypes.object.isRequired,
  pageType: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default LoadingError;
