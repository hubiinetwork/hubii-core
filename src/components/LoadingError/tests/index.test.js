import React from 'react';
import { shallow } from 'enzyme';

import LoadingError from '../index';

describe('<LoadingError />', () => {
  it('should render correctly when unknown error message passed', () => {
    const wrapper = shallow(<LoadingError id={'12'} error={{ message: 'foo' }} pageType={'bar'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when Not Found error message is passed', () => {
    const wrapper = shallow(<LoadingError id={'12'} error={{ message: 'Not Found' }} pageType={'bar'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
