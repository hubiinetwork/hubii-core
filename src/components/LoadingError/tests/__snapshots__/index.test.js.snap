// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingError /> should render correctly when Not Found error message is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingError
    error={
        Object {
            "message": "Not Found",
          }
    }
    id="12"
    pageType="bar"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FormattedMessage
        defaultMessage="{type} {id} doesn't exist"
        id="app.containers.LoadingError.doesntExist"
        values={
                Object {
                        "id": "12",
                        "type": "bar",
                      }
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultMessage": "{type} {id} doesn't exist",
        "id": "app.containers.LoadingError.doesntExist",
        "values": Object {
          "id": "12",
          "type": "bar",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FormattedMessage
          defaultMessage="{type} {id} doesn't exist"
          id="app.containers.LoadingError.doesntExist"
          values={
                    Object {
                              "id": "12",
                              "type": "bar",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMessage": "{type} {id} doesn't exist",
          "id": "app.containers.LoadingError.doesntExist",
          "values": Object {
            "id": "12",
            "type": "bar",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<LoadingError /> should render correctly when unknown error message passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingError
    error={
        Object {
            "message": "foo",
          }
    }
    id="12"
    pageType="bar"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FormattedMessage
        defaultMessage="Error occured: {error}"
        id="app.containers.LoadingError.unknownError"
        values={
                Object {
                        "error": "foo",
                      }
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultMessage": "Error occured: {error}",
        "id": "app.containers.LoadingError.unknownError",
        "values": Object {
          "error": "foo",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FormattedMessage
          defaultMessage="Error occured: {error}"
          id="app.containers.LoadingError.unknownError"
          values={
                    Object {
                              "error": "foo",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultMessage": "Error occured: {error}",
          "id": "app.containers.LoadingError.unknownError",
          "values": Object {
            "error": "foo",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
